### ISSUE TITLE
[.NET 10] certain `[Obsolete]` warnings are not reported

### ISSUE THREAD

**Issue by jonathanpeppers** [dotnet/roslyn#76226]
**Version Used**:  .NET SDK 10.0.100-alpha.1.24573.1

We noticed this from Maestro/codeflow here:

* https://github.com/dotnet/android/pull/9395

**Steps to Reproduce**:

1. Call `AppDomain.CreateDomain("test")` in .NET 10
2. Note that you don't get a `SYSLIB0024` warning

Repro: [testroslyn.zip](https://github.com/user-attachments/files/17981183/testroslyn.zip)

Note that this seems to work fine using `net9.0` and a .NET 9 SDK.

**Diagnostic Id**: 

```
SYSLIB0024: 'AppDomain.CreateDomain(string)' is obsolete: 'Creating and unloading AppDomains is not supported and throws an exception.'
```

**Expected Behavior**:

I get a `SYSLIB0024` warning.

**Actual Behavior**:

I get no `SYSLIB0024` warning.

**Other info**:

Note that if I open `packs\Microsoft.NETCore.App.Ref\10.0.0-alpha.1.24570.9\ref\net10.0\System.Runtime.dll`, it looks correct:

![Image](https://github.com/user-attachments/assets/fec4aa6f-922b-4070-a963-34a4a6c3702b)

The warning also seems to work fine if I declare an `[Obsolete]` method within the C# file.

**Comment by CyrusNajmabadi** [comment]
I don't know what SYSLIB is.  But is isn't roslyn :)

**Comment by jonathanpeppers** [comment]
@CyrusNajmabadi this is if you have `[Obsolete(DiagnosticId="` and put `SYSLIB0024`. Maybe it is any `[Obsolete]` in a reference assembly?

**Comment by jonathanpeppers** [comment]
Ok, this works fine, I get a warning if I call it:
```csharp
[Obsolete]
public void Test()
{
}
```

This doesn't produce a warning at all:
```csharp
[Obsolete("whoopsies", DiagnosticId = "SYSLIB0024")]
public void Test()
{
}
```

**Comment by AlekseyTs** [comment]
Is it possible that "SYSLIB0024" warning is disabled by default in the new SDK?

@jonathanpeppers What happens if you use a different diagnostic id? For example:
```
[Obsolete("whoopsies", DiagnosticId = "CustomID_0024")]
public void Test()
{
}
```

**Comment by jonathanpeppers** [comment]
When testing this, I found this also produces no warnings:
```csharp
[Obsolete("whoopsies")]
public void Test()
{
}
```

What made me notice this issue, warnings related to `[SupportedOSPlatformVersion]` on Android were also not working in .NET 10. I was hoping whatever the underlying issue was maybe fixed both problems here. `[Obsolete]` seems more fundamental and reproducible in a console app, though.

**Comment by Youssef1313** [comment]
I investigated this. It's not a Roslyn bug, but an SDK bug. WarningLevel used to align with the .NET version, and the way the SDK does it is using this:

```
    <!-- .NET projects, however, can float up to their TFM's major version -->
    <WarningLevel Condition="'$(Language)' == 'C#' And '$(WarningLevel)' == '' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' ">$(_TargetFrameworkVersionWithoutV.Substring(0, 1))</WarningLevel>

```

Now we are two digit version (10) 😄 
So, this trick no longer works.

**Comment by Youssef1313** [comment]
This is fixed now in SDK and can be closed

**Comment by jonathanpeppers** [comment]
When Maestro/codeflow makes it to the dotnet/android repo, I'll be able to try it out and close this, thanks!

**Comment by AlekseyTs** [comment]
@jonathanpeppers I am going to close this issue as fixed. If the problem is still there with updated SDK, please reactivate this issue and attach a complog (https://github.com/jaredpar/complog).
